lib/metasploit/framework/password_crackers/jtr/formatter.rb

```diff
@@ -79,10 +82,47 @@ def hash_to_jtr(cred)
       # /ssha/
       # /raw-sha512/
       # /raw-sha256/
+      # /xsha/
+      # /mscash2/
       # This also handles *other* type credentials which aren't guaranteed to have a public
 
       return "#{cred.public.nil? ? ' ' : cred.public.username}:#{cred.private.data}:#{cred.id}:"
     end
   end
   nil
 end
+
+# This method takes a {framework.db.cred}, and normalizes it
+# from the JTR format to the DB format.
+#
+# @param [credClass] a credential from framework.db
+# @return [Array] All of the hash types that may be in the DB that apply
+def jtr_to_db(cred_type)
```

Comment:
I think this came up before in a different PR; but it'd be great to namespace this properly behind a mixin or similar to ensure it's not a global method 😄

Reply:
Fixed, also did another rebase since its been a few months